### PR TITLE
Some tweaks to our DLC message handling

### PR DESCRIPTION
**Pull request by luckysori**
I noticed this when I was looking into #410.

Sorry about the last two patches effectively being fixups of the first one. I chose to leave it like this because rebasing was annoying.

I'm pretty sure it works, but I wasn't able to test it in full because of #438.

### CODE REVIEW

File: mobile/native/src/ln_dlc/node.rs [hunk 1]
```diff
@@ -82,151 +85,156 @@ impl Node {
...
+                    // TODO: We should probably verify that: (1) the counterparty is the
+                    // coordinator and (2) the DLC channel close offer is expected and correct.
+                    self.inner
+                        .accept_dlc_channel_collaborative_settlement(&channel_id)
+                        .with_context(|| {
+                            format!(
+                                "Failed to accept DLC channel close offer for channel {}",
+                                hex::encode(channel_id)
+                            )
+                        })?;
```
Comment (1010Tom) [inline mobile/native/src/ln_dlc/node.rs]: Shouldn't this be a match block?

File: mobile/native/src/ln_dlc/node.rs [hunk 2]
```diff
@@ -174,28 +173,32 @@ impl Node {
 
                     match reply_msg {
                         SubChannelMessage::Finalize(SubChannelFinalize { channel_id, .. }) => {
-                            let contract = self
+                            let contracts = self
                                 .inner
                                 .dlc_manager
                                 .get_store()
-                                .get_contract(&channel_id)
```
Comment (klochowicz) [inline mobile/native/src/ln_dlc/node.rs]: too bad these were only type aliases, because such a API misuse seems to be quite common :(

**Review by klochowicz — APPROVED** [review]
LGTM!

File: mobile/native/src/ln_dlc/node.rs (hunk 2 above)
Comment (luckysori) [reply mobile/native/src/ln_dlc/node.rs]: Definitely an insidious source of bugs. We could PRs on the relevant repos to introduce new types. I reckon `rust-dlc` would accept it.

File: mobile/native/src/ln_dlc/node.rs (hunk 1 above)
Comment (luckysori) [reply mobile/native/src/ln_dlc/node.rs]: For reference: the `if let`s were already there, I just moved them around.

I do agree that a `match` is a bit clearer as one can more easily see that the cases are independent. On the other hand, it does mean that we have to explicitly handle the catch-all pattern explicitly, which is a bit noisy.

Comment (luckysori) [reply mobile/native/src/ln_dlc/node.rs]: Improved with https://github.com/get10101/10101/pull/439/commits/412605a2af653877ac1a64510aeae1d8a95c2a4e.

**Pull request merged**